build-tools/packages/bundle-size-tools/package.json

```diff
@@ -50,6 +50,7 @@
 	"devDependencies": {
 		"@biomejs/biome": "~1.8.3",
 		"@fluidframework/build-common": "^2.0.3",
+		"@fluidframework/build-tools": "workspace:~",
```

Comment:
This does not look correct. Everything in build-tools should only use prior versions of build-tools (root of build-tools workspace) for building. Otherwise, we need to be very careful. I think it is a reasonable limitation.
We can install prior versions to get their bins with an alias. The policy doesn't understand aliases though.
Policy (tooling) fix: #21883 
Dependency update (bump and alias use): #21884

Reply:
I agree it's weird to use the in-workspace deps. But it's also odd to use the previous version or to be inconsistent between packages in the workspace.

Maybe the policy should be updated to be more configurable. In this case just to ignore build-tools as a missing dep instead of having to ignore the whole policy.

Reply:
@jason-ha what exactly is the concern if it uses the workspace version to build itself? I could even see a bit of an advantage: if we break it (bad enough, at least), the release group itself won't build so we won't be able to release it. I can imagine there might be pitfalls too but I can't see them clearly.

Reply:
There are two concerns. But first let me say that the pattern to use previous tools is well-understood. Compilers are commonly built like this. Consider the C compilers. C compiler can be written in C but only after a version of C compiler exists. First C compiler was not written in C, but later were and they could not depend on themselves. So, they depended on a previous version of same tool.
build-tools workspace is like that.
Concern 1: build-tools package obviously can't depend on itself to build itself or any of the packages it has production dependencies on, but it can use a prior version. I think, the workspace has scripts that allow a build without using build-tools package. If so, one could be careful and then maybe use just built build-tools in later builds. But there are questions about module loading order and rewriting. It just is more complicated than we care to think about for the value.
Concern 2: Other packages that are separate from build-tools package and its prod deps, could depend on build-tools if we have ensured that build-tools package is fully built before its use. The dependency system is not built for that AND we must already be using a different version of build-tools to even evaluate dependencies. Better to stay consistent.
That said, if you wanted to be careful and test local tool changes within build-tools workspace, then I think that could be done. It just shouldn't be the default. It would be really nice to have more test coverage and setups.